services/stip_relayer/main.go

```diff
@@ -0,0 +1,434 @@
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err() // exit if context is cancelled
+		case <-ticker.C:
+			if err := s.ProcessExecutionResults(ctx); err != nil {
+				// Log the error and decide whether to continue based on the error
+				fmt.Printf("Error processing execution results: %v", err)
+				// Optionally, you can return the error to stop the goroutine
+				// return err
+			}
+		}
+	}
+}
+
+// processExecutionResults encapsulates the logic for requesting and storing execution results
+func (s *STIPRelayer) ProcessExecutionResults(ctx context.Context) error {
+	resp, err := ExecuteDuneQuery()
+	if err != nil {
+		return fmt.Errorf("failed to execute Dune query: %v", err)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("failed to read response body: %v", err)
+	}
+
+	var result map[string]string
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal response body: %v", err)
+	}
+
+	executionID, ok := result["execution_id"]
+	if !ok {
+		return fmt.Errorf("no execution_id found in response")
+	}
+
+	time.Sleep(20 * time.Second) // Consider replacing this with a more robust solution
+
+	executionResults, err := GetExecutionResults(executionID)
+	if err != nil {
+		return fmt.Errorf("failed to get execution results: %v", err)
+	}
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("expected status code 200, got %d", resp.StatusCode)
+	}
+
+	getResultsBody, err := ioutil.ReadAll(executionResults.Body)
+	var jsonResult QueryResult
+	err = json.Unmarshal(getResultsBody, &jsonResult)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling JSON: %v", err)
+	}
+	fmt.Println(jsonResult.Result.Rows)
+	fmt.Println("Number of rows:", len(jsonResult.Result.Rows))
+
+	// Convert each Row to a STIPTransactions and store them in the database
+	return s.StoreResultsInDatabase(ctx, jsonResult.Result.Rows, jsonResult.ExecutionID)
+}
+
+// storeResultsInDatabase handles the storage of results in the database
+func (s *STIPRelayer) StoreResultsInDatabase(ctx context.Context, rows []Row, executionID string) error {
+	stipTransactions := make([]db.STIPTransactions, len(rows))
+	for i, row := range rows {
+		stipTransactions[i] = db.STIPTransactions{
+			Address:     row.Address,
+			Amount:      row.Amount,
+			AmountUSD:   row.AmountUsd,
+			ArbPrice:    row.ArbPrice,
+			BlockTime:   row.BlockTime.Time,
+			Direction:   row.Direction,
+			ExecutionID: executionID,
+			Hash:        row.Hash,
+			Module:      row.Module,
+			Token:       row.Token,
+			TokenPrice:  row.TokenPrice,
+			Rebated:     false,
+		}
+	}
+
+	if len(stipTransactions) > 0 {
+		if err := s.db.InsertNewStipTransactions(ctx, stipTransactions); err != nil {
+			return fmt.Errorf("error inserting new STIP transactions: %v", err)
+		}
+	}
+
+	// Optionally, confirm that the insert occurred
+	stipTransactionsNotRebated, err := s.db.GetSTIPTransactionsNotRebated(ctx)
```

Comment:
Ideally would have this be its own func

Reply:
removed read query after insert for no reason